apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx

```diff
@@ -374,6 +375,36 @@ export class DetailViewContents extends React.Component {
     });
   };
 
+  handleReopenApplicationClick = () => {
```

Comment:
I found it challenging to write this code "well" without doing a lot of refactoring first. Any suggestions for how to clean it up are welcome